miden-lib/asm/miden/kernels/tx/tx.masm

```diff
@@ -176,3 +319,36 @@ export.create_note
     dup movdn.5 exec.memory::set_created_note_recipient
     # => [note_ptr, 0, 0, 0, 0, 0, 0, 0, 0]
 end
+
+#! Adds an ASSET to the specified note and returns a pointer to the memory address
+#! at which the note is stored.
+#!
+#! Inputs: [note_ptr, ASSET]
+#! Outputs: [note_ptr]
+#!
+#! ptr is the pointer to the memory address at which the note is stored.
+#! ASSET can be a fungible or non-fungible asset.
+export.add_asset_to_note
+    # validate the ASSET
+    movdn.4 exec.asset::validate_asset
+    # => [ASSET, note_ptr]
+
+    # emit the event (I need to push.0 drop for the decorator to work)
```

Comment:
minor nit: I would not use the first person singular subject in comments - e.g., I'd say: "we need to..." instead of "I need to...").